From 94faa241c1c88ff33e654654df54132b70141bfd Mon Sep 17 00:00:00 2001
Message-Id: <patch-18.24-94faa241c1c-20220309T121511Z-avarab@gmail.com>
In-Reply-To: <cover-00.24-00000000000-20220309T121511Z-avarab@gmail.com>
References: <cover-00.24-00000000000-20220309T121511Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Sun, 27 Feb 2022 20:56:32 +0100
Subject: [PATCH 18/24] revisions API: have release_revisions() release
 "grep_filter"
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Extend the the release_revisions() function so that it frees the
"grep_filter" in the "struct ref_info".

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 revision.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/revision.c b/revision.c
index 6c83a715dee..f51a60b2a96 100644
--- a/revision.c
+++ b/revision.c
@@ -2953,6 +2953,7 @@ void release_revisions(struct rev_info *revs)
 	object_array_clear(&revs->pending);
 	release_revisions_cmdline(&revs->cmdline);
 	release_revisions_mailmap(revs->mailmap);
+	free_grep_patterns(&revs->grep_filter);
 }
 
 static void add_child(struct rev_info *revs, struct commit *parent, struct commit *child)
-- 
2.35.1.1295.g6b025d3e231

